### DISCUSSION TITLE
Touch effect in a style

### DISCUSSION THREAD

**Discussion by developer9969** [xamarin/XamarinCommunityToolkit#1045] [General]
Hi can touch effect be used in a style, 
I have a base button that is defined in resource dictionary

		<Style x:Key="ButtonStyle" TargetType="buttons:SfButton">
			<Setter Property="BackgroundColor" Value="{DynamicResource ButtonBackgroundColor}"></Setter>
			<Setter Property="CornerRadius" Value="20"></Setter>    
		</Style>

Can I move below properties to a style above

		xct:TouchEffect.AnimationDuration="250"
		xct:TouchEffect.AnimationEasing="{x:Static Easing.CubicInOut}"
		xct:TouchEffect.PressedScale="0.8"
		xct:TouchEffect.PressedOpacity="0.6"

Also just checking the nuget must be added to both android and iOS project , will it work is added just to the shared project?

Many thanks

**Comment by AndreiMisiukevich** [comment]
Hello @developer9969 
As for the second question: No it won't work if you add XCT nuget only to the shared project. You must add it to platform-specific projects as well.

**Reply by AndreiMisiukevich** [reply]
Yes, you can add those properties to styles

```
        <Style x:Key="StyleName" TargetType="StackLayout">
            <Setter Property="xct:TouchEffect.PressedScale" Value="0.8" />
            <Setter Property="xct:TouchEffect.PressedOpacity" Value="0.6" />
        </Style>
```

**Reply by developer9969** [reply]
Thanks for replying to both questions.All Good.

**Comment by developer9969** [comment]
@AndreiMisiukevich  hi there not sure if is a resharper issue but I get the following error in style. However it compile and works.

![image](https://user-images.githubusercontent.com/16970109/110622925-068d5280-8194-11eb-9b00-a00fd2f14000.png)

many thanks

**Reply by AndreiMisiukevich** [reply]
Hmm, need to ask somebody on windows to verify it.
It works pretty fine on VS for Mac at least

CC @pictos

**Reply by pictos** [reply]
On it!

**Reply by pictos** [reply]
@developer9969 @AndreiMisiukevich 
Worked for me (the build was succedd), but since we're on `App.xaml` the XCT xmlns should be:
`xmlns:xct="clr-namespace:Xamarin.CommunityToolkit.Effects;assembly=Xamarin.CommunityToolkit"`

**Reply by developer9969** [reply]
Hi I can confirm that is a resharper issue my collegue on windows too without resharper does not get those errors, still compiles and build and very happy with touch effect, annoying about resharper though
